sdk/dotnet/Pulumi/Serialization/Attributes.cs

```diff
@@ -96,14 +96,15 @@ public sealed class EnumTypeAttribute : Attribute
     }
     
     [AttributeUsage(AttributeTargets.Class)]
-    public sealed class ResourceIdentifierAttribute : Attribute
+    public sealed class ResourceTypeAttribute : Attribute
     {
-        public string Token { get; }
+        public string Type { get; }
+        
         public string? Version { get; }
 
-        public ResourceIdentifierAttribute(string token, string? version)
+        public ResourceTypeAttribute(string type, string? version = null)
```

Comment:
Nit: Should we remove the `version` parameter and change the `Version` property to have a setter, to align with the Framework Design Guidelines?

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/attributes

> ❌ AVOID providing constructor parameters to initialize properties corresponding to the optional arguments.
> 
> In other words, do not have properties that can be set with both a constructor and a setter. This guideline makes very explicit which arguments are optional and which are required, and avoids having two ways of doing the same thing.

Though, not sure if this guideline has evolved at all since the C# optional arguments feature was introduced.

I don't feel strongly about this one way or the other, just wanted to point it out. I'm good with keeping it as-is.

Reply:
I actually made `version` a required (but nullable) parameter